test/browsers.test.ts

```diff
@@ -23,112 +24,122 @@ describe('connect flow', () => {
     expect(frame).toBeDefined()
   })
 
-  test('create account', async () => {
-    // Ensure 3ID Connect iframe is present
-    const frame = await page.frame('threeid-connect')
-
-    // Run account creation flow in page
-    const accountCreatedPromise = page.evaluate(() => {
-      const provider = window.createEthereumAuthProvider()
-      return window.threeIdConnect.connect(provider).then(() => {
-        return window.threeIdConnect.createAccount().then(() => {
-          return provider.accountId().then((id) => id.toString())
-        })
+  PROVIDERS.forEach((providerType) => {
+    describe(providerType, () => {
+      test('create account', async () => {
+        // Ensure 3ID Connect iframe is present
+        const frame = await page.frame('threeid-connect')
+
+        // Run account creation flow in page
+        const accountCreatedPromise = page.evaluate((type) => {
+          return window.createAuthProvider(type).then((provider) => {
+            return window.threeIdConnect.connect(provider).then(() => {
+              return window.threeIdConnect.createAccount().then(() => {
+                return provider.accountId().then((id) => id.toString())
+              })
+            })
```

Comment:
Why are we using `.then`rather than await?

Reply:
The code in `evaluate` is injected in the browser and evaluated there, so from my understanding there are restrictions to what can be injected, and `async`/`await` statements don't seem to be handled.